src/v/cloud_storage/remote_segment_index.cc

```diff
@@ -345,9 +350,8 @@ void remote_segment_index_builder::consume_batch_start(
   model::record_batch_header hdr,
   size_t physical_base_offset,
   size_t size_on_disk) {
-    if (
-      hdr.type == model::record_batch_type::raft_configuration
-      || hdr.type == model::record_batch_type::archival_metadata) {
+    auto it = std::find(_filter.begin(), _filter.end(), hdr.type);
+    if (it != _filter.end()) {
```

Comment:
I'm good with this change, but looking at the impl for raft::offset_translator_batch_types, why wasn't it implemented to be the list of batches that increase the kafka offset? Seems like such a list would never change, and it seems easy to forget to add to the current list

E.g. just looking at the current list, should `group_*` be accounted for? I know we don't upload the consumer groups topic as is, but I don't think we _disallow_ uploading

Reply:
Such list may also change, I believe. Some new transactional batch type can be added or something like this.
The 'group_' batches are not participating in offset translation I believe.